wp-includes/js/service-worker-offline-commenting.js

```diff
@@ -0,0 +1,82 @@
+/* global ERROR_OFFLINE_URL, ERROR_MESSAGES, ERROR_500_URL */
+{
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'wpPendingComments' );
+	const errorMessages = ERROR_MESSAGES;
+
+	const commentHandler = ( { event } ) => {
+
+		const clone = event.request.clone();
+		return fetch( event.request )
+			.then( ( response ) => {
+				if ( response.status < 500 ) {
+					return response;
+				}
+
+				// @todo Replace depending on BroadcastChannel.
+				const channel = new BroadcastChannel( 'wordpress-server-errors' );
```

Comment:
I added currently the logic here without creating global methods (yet) to see what's needed exactly. Also removed the streaming from here since it didn't seem to be necessary in case of `wp-comments-post`.

Reply:
That's right. No need to do streaming in handling `wp-comments-post.php` requests. It should be redirecting, or serving the offline page in its entirety.